mailpoet/lib/Newsletter/Renderer/Blocks/Button.php

```diff
@@ -7,6 +7,7 @@
 
 class Button {
   public function render($element, $columnBaseWidth) {
+    $originalWidth = $element['styles']['block']['width'];
     $element['styles']['block']['width'] = $this->calculateWidth($element, $columnBaseWidth);
```

Comment:
Seeing this change, I noticed that the width is processed in `Button::calculateWidth` and it takes into account also column width. The method ensures that the resulting width is not bigger than the column.
So when we use the $originalWidth as is and place button in the three column design the wrapping table is wider than the column.

![Screenshot 2024-07-23 at 16 04 55](https://github.com/user-attachments/assets/9f303ac7-c793-4f69-af54-8be5b2748694)

I didn't observe any rendering issue with that, but I would prefer we fix that to be safe. What do you think?

Reply:
@costasovo Good catch. It's addressed in [the latest commit](https://github.com/mailpoet/mailpoet/pull/5745/commits/de0984ae2c1145a4b7b40d9f49f62e27577ceec6).